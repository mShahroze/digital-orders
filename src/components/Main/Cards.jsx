import React, { Component } from 'react';
import Card from './CardUI';
import img1 from '../assets/126561.jpg'


export class Cards extends Component {

  state = {
    blurCard: false,
    cardData: [{ id: 1, hovered: false }, { id: 2, hovered: false }, { id: 3, hovered: false }, { id: 4, hovered: false }],
  }

  onCardHover = (card) => {
    let newCardData = this.state.cardData
    newCardData.map(c => {
      return c.id === card.id ? c.hovered = true : c.hovered = false
    })
    this.setState({
      blurCard: true,
      cardData: newCardData
    })
  }

  onCardLeave = () => {
    let newCardData = this.state.cardData
    newCardData.map(c => {
      return c.hovered = false
    })
    this.setState({
      blurCard: false,
      cardData: newCardData
    })
  }

  render() {

    return (
      <div className="container">
        <div className="cards row">
          {this.state.cardData.map(card => {
            return <div key={card.id} className={this.state.blurCard === true && card.hovered === false ? "cardBlur" : "cardNoBlur"} onMouseEnter={() => this.onCardHover(card)} onMouseLeave={() => this.onCardLeave(card)}>
              <Card imgsrc={img1} />
            </div>
          })}
        </div>
      </div>
    )
  }
}

export default Cards
