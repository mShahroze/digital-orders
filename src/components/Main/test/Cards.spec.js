import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Cards } from '../Cards'

Enzyme.configure({ adapter: new Adapter() });

describe('<Cards />', () => {
  let wrapper;
  let instance
  beforeEach(() => {
    wrapper = mount(<Cards />)
    instance = wrapper.instance();
  });

  test('Snapshot is correct', () => {
    expect(wrapper).toMatchSnapshot();
  })


  test('onCardHover', () => {
    const card = { id: 1, hovered: false };

    instance.onCardHover(card);

    expect(instance.state.blurCard).toEqual(true);
    instance.state.cardData.forEach((stateCard) => {
      if (stateCard.id === 1) {
        expect(stateCard.hovered).toEqual(true);
      }
      else {
        expect(stateCard.hovered).toEqual(false);
      }
    });
  });
});
