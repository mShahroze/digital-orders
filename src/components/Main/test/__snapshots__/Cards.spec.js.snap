// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards /> Snapshot is correct 1`] = `ReactWrapper {}`;
